lint_rules/lint_rules/debug_rules.py

```diff
@@ -284,13 +303,24 @@ def fix_subroutine(cls, subroutine, rule_report, config):
                 vtype = arg.type.clone(shape=new_shape, scope=subroutine)
                 new_vars += as_tuple(arg.clone(type=vtype, dimensions=new_shape, scope=subroutine))
 
-        #TODO: add 'VariableDeclaration.symbols' should be of type 'Variable' rather than 'Expression'
+        # simplify variable declarations
+        single_variable_declaration(subroutine)
```

Comment:
Ok, apologies if I'm misreading this, but from what I understand, fix_subroutine is the auto that is intended to "correct" small violations for code that gets put back into source, right? With that in mind, splitting variable declarations looks like a very invasive change for a linter! I know we like these for GPU-specific transformations, but forcing this into every file of a large code base for mere convenience seems like a pretty hard no, if we want to bring "fixed" code back into any repo. Is this really intended here, or am I misunderstanding this?

Reply:
Your understanding is correct....this was an easy yet robust way to deal with any potential corner case of array declarations. I thought it would be ok given how invasive this fixer method is already.

Thanks a lot for pointing this out, I'll stop being lazy and make the fixer method more sophisticated 😅